routes/api/product_type.js

```diff
@@ -31,5 +31,72 @@ router.get("/get-all", async (req, res) => {
 		res.json(q);
 	}
 });
+// Validation
+const productValidation = [
+	check("type_name")
+		.not()
+		.isEmpty()
+		.withMessage("Type Name cannot be empty."),
+];
+
+// 2.
+router.post("/add", productValidation, async (req,res) => {
+	const validationErrors = validationResult(req);
+	if (!validationErrors.isEmpty()) {
+		return res.status(422).json({ errors: validationErrors.array() });
+	}
+	const { type_name } = req.body;
+	const q = await utils.insert({
+		table: "product",
+		info: {
+			type_name
+		},
+		returning: "type_name"
+	})
+	if (q.errors) {
+		res.status(500).json(q);
+	} else {
+		res.json(q);
+	}
+});
+
+// 3.
+router.put("/:product_code/edit", productValidation, async (req,res) => {
+	const validationErrors = validationResult(req);
+	if (!validationErrors.isEmpty()) {
+		return res.status(422).json({ errors: validationErrors.array() });
+	}
+	const { type_name } = req.body;
```

Comment:
This one is a bit tricky.
You have to receive from both the ```req.body``` (contains the edited name) and ```req.params``` (contains the name being edited).
Try
```
const { type_name } = req.body;
const { type_name: type_name_params} = req.params;
```
Then in where:
```where: `"type_name" = '${type_name_params}'`, ```